.github/prune_old_docs.py

```diff
@@ -0,0 +1,57 @@
+# This script is run by release-please "change-versions.yml"
+# from the root of the docs folder
+
+import json
+import os
+import shutil
+
+from collections import namedtuple
+
+VERSIONS_FILE_NAME = "versions.json"
+
+Version = namedtuple("Version", "major minor patch")
+
+
+def main():
+    versions_to_keep = []
+    versioned_docs_to_delete = []
+    versioned_sidebars_to_delete = []
+    with open(VERSIONS_FILE_NAME, "r") as available_versions:
+        available_versions_json = json.loads(available_versions.read())
+        current_version = str_to_version(available_versions_json[0])
+        for version_str in available_versions_json:
+            version = str_to_version(version_str)
+            # if the version being processed is the same as the current major and minor version do nothing
+            if version.major == current_version.major and version.minor == current_version.minor:
+                versions_to_keep.append(version_str)
+                continue
+
+            # if the version is being processed is "0" patch then just keep it
+            if version.patch == "0":
```

Comment:
Why do we keep patch `0` but not patch `max_patch`

Reply:
0 would be consistent across versions and easier on eyes. but I can do this if you feel strongly about it?

Reply:
I would imagine that we want to keep the latest version of each documentation for each `major` and `minor`

Reply:
Works for me 👍 Will change the script a bit more :)

Reply:
<img width="1728" alt="image" src="https://user-images.githubusercontent.com/6670312/233150671-4f86dc71-9d25-420b-bbd5-2e0d2124a18f.png">